ext/cl_khr_command_buffer_mutable_dispatch.asciidoc

```diff
@@ -0,0 +1,980 @@
+  the queue when _command_ was recorded, then the queue associated with the
+  command-buffer that _command_ belongs to is returned.
+
+| `CL_MUTABLE_COMMAND_COMMAND_BUFFER_KHR`
+| `cl_command_buffer_khr`
+| Return the command-buffer associated with _command_.
+
+| `CL_MUTABLE_COMMAND_COMMAND_TYPE_KHR`
+| `cl_command_type`
+| Return the command-type associated with _command_.
+
+  The list of supported event command types defined by *clGetEventInfo* is used
+  with the matching command.
+
+| `CL_MUTABLE_DISPATCH_PROPERTIES_ARRAY_KHR`
+| `cl_ndrange_kernel_command_properties_khr[]`
+| Return the properties argument specified on _command_ recording with
+  *clCommandNDRangeKernel*.
+
+  If the properties argument specified on creation of _dispatch_ was not
+  `NULL`, the implementation must return the values specified in the
+  properties argument in the same order and without including additional
+  properties.
+
+  If the properties argument specified on creation of _dispatch_ was `NULL`,
+  or _command_ was not recorded from a *clCommandNDRangeKernel* command, the
+  implementation must return _param_value_size_ret_ equal to 0, indicating that
+  there are no properties to be returned.
+
+| `CL_MUTABLE_DISPATCH_KERNEL_KHR`
+| `cl_kernel`
+| Return the kernel associated with _command_ when recorded with
+  *clCommandNDRangeKernel*.
+
+  If _command_ was not recorded from a *clCommandNDRangeKernel* command, the
+  implementation must return _param_value_size_ret_ equal to 0, indicating that
+  the value returned in _param_value_ is not valid.
+
+| `CL_MUTABLE_DISPATCH_DIMENSIONS_KHR`
+| `cl_uint`
+| Return the number of work-item dimensions specified when _dispatch_ was
+  created.
+
+  If _command_ was not recorded from a *clCommandNDRangeKernel* command, the
+  implementation must return _param_value_size_ret_ equal to 0, indicating that
+  the value returned in _param_value_ is not valid.
+
+| `CL_MUTABLE_DISPATCH_GLOBAL_WORK_OFFSET_KHR`
+| `size_t[]`
+| Return the global work-item offset set on _dispatch_ creation, or from
+  the most recent update via *clUpdateMutableCommandsKHR* where this value
+  was modified. The output array contains _work_dim_ values, where _work_dim_ is
+  returned by the query `CL_MUTABLE_DISPATCH_DIMENSIONS_KHR`. If a global work-item
+  offset was not set, zero is returned for each element in the array.
+
+  If _command_ was not recorded from a *clCommandNDRangeKernel* command, the
+  implementation must return _param_value_size_ret_ equal to 0, indicating that
+  the value returned in _param_value_ is not valid.
+
+| `CL_MUTABLE_DISPATCH_GLOBAL_WORK_SIZE_KHR`
+| `size_t[]`
+| Return the global work-item size set on _dispatch_ creation, or from
+  the most recent update via *clUpdateMutableCommandsKHR* where this value
+  was modified. The output array contains _work_dim_ values, where _work_dim_ is
+  returned by the query `CL_MUTABLE_DISPATCH_DIMENSIONS_KHR`.
```

Comment:
```suggestion
  returned by the query `CL_MUTABLE_DISPATCH_DIMENSIONS_KHR`.
  If _command_ was created with a `NULL` value for _global_work_size_
  then zero is returned for each element in the array.
```

Or, alternative wording more consistent with the other table entries:

```suggestion
  returned by the query `CL_MUTABLE_DISPATCH_DIMENSIONS_KHR`.
  If a global work-item size was not set, zero is returned for each element in
  the array.
```

Sorry, this isn't technically editorial, but I don't see what else you would do in this case.

`clCommandNDRangeKernelKHR` inherits the rules for `global_work_size` from `clEnqueueNDRangeKernel`, where in OpenCL 2.0+ 0's are allowed, and NULL is the same as a 0.

Note that wording does not apply to changes to the value by `clUpdateMutableCommandsKHR` because this extension says

> If global_work_size is NULL then the number of global work-items in the dispatch is not changed.

I don't think the difference is a bug, I think it's different default behaviour for different situations.

Reply:
hmm well spotted, I can't think of a way to resolve that ambiguity either. Only that we could provide a note to the reader that such a situation may occur.